### ISSUE TITLE
status.podIP empty for HostNetwork tasks

### ISSUE THREAD

**Issue by djosborne** [kubernetes/kubernetes#43531]
**Kubernetes version**: v1.6.0-beta.4

I think this may be the same issue as #39113, but I'm opening separately since I'm seeing the behavior for pods in hostNetwork (instead of a CNI networked pod).

**What happened**: 

I have a pod spec with this snippet of yaml:
```
hostNetwork: true
containers:
  spec:
    env:
      - name: ETCD_IP
        valueFrom:
          fieldRef:
            fieldPath: status.podIP
    command: ["/bin/sh","-c"]
    args: ["/usr/local/bin/etcd --advertise-client-urls=http://$ETCD_IP:6666 --name=calico --data-dir=/var/etcd/calico-data --listen-client-urls=http://0.0.0.0:6666 --listen-peer-urls=http://0.0.0.0:6667"]


```

In v1.5.2, my Pod is coming up with `$ETCD_IP` correctly filled in to the node's IP.

In 1.6.0-beta.4, `$ETCD_IP` is empty at startup. 

The containerStatus in the pod yaml _does_ correctly shows the node's IP in the podIP field:
```
  containerStatuses:
  - containerID: docker://42b77d3538c71c7d8aba28321d5b611c05ed274863676006187de3077f30c16e
    image: gcr.io/google_containers/etcd:2.2.1
    imageID: docker-pullable://gcr.io/google_containers/etcd@sha256:19544a655157fb089b62d4dac02bbd095
f82ca245dd5e31dd1684d175b109947
    lastState: {}
    name: calico-etcd
    ready: true
    restartCount: 0
    state:
      running:
        startedAt: 2017-03-22T17:08:35Z
  hostIP: 10.240.0.31
  phase: Running
  podIP: 10.240.0.31
  qosClass: BestEffort
  startTime: 2017-03-22T17:08:32Z
```

I heard K8s 1.6 moved a lot of core networking into separate CNI plugins. Does this mean that hostNetwork is now handled by a CNI plugin, and would hit the same bug as #39113?

**Comment by calebamiles** [comment]
Could someone from @kubernetes/sig-node-bugs or @kubernetes/sig-network-bugs please help triage this issue and see whether it's covered by https://github.com/kubernetes/kubernetes/issues/39113? Thanks!

**cc: @ethernetdan, @kubernetes/kubernetes-release-managers**

**Comment by dchen1107** [comment]
@bowei could you please triage or delegate this one to others? Thanks!

**Comment by bowei** [comment]
@yujuhong I think has an update on this

**Comment by thockin** [comment]
@pmorie because fingerprints

I do not think this was changed on purpose.

**Comment by yujuhong** [comment]
Confirmed that this is a regression in the CRI implementation. Looking into the code now.

**Comment by yujuhong** [comment]
Will send a fix shortly.

**Comment by yujuhong** [comment]
The fix #43539 has been merged.
